slack.cpp

```diff
@@ -151,4 +152,27 @@ void Slack::reconfigure(const string& newConfig)
 	ConfigCategory category("new", newConfig);
 	m_url = category.getValue("webhook");
 	m_text = category.getValue("text");
+	verifyURLFormat();
+}
+
+/**
+ * Verify Slack webhook has valid URL format
+ *
+ */
+void Slack::verifyURLFormat()
```

Comment:
Remove this; checking that its a URL and not empty after trim is enough.

Otherwise this may cause issues if GovSlack is used; or in future Slack allows self hosted enterprise solutions or custom domains.

Reply:
Also raise a Jira for URL semantics vs current limitation for an item of type URL in config.

Reply:
Since type URL in config will check not allow blank value to be saved as it's not valid URL. Also mandatory : true is used so it webhook is required field now.